dev/ci/gitlab.bat

```diff
@@ -29,10 +32,23 @@ call %CI_PROJECT_DIR%\dev\build\windows\MakeCoq_MinGW.bat -threads=1 ^
   -arch=%ARCH% -installer=Y -coqver=%CI_PROJECT_DIR_CFMT% ^
   -destcyg=%CYGROOT% -destcoq=%DESTCOQ% -cygcache=%CYGCACHE% ^
   -addon="bignums ltac2 equations" -make=N ^
-  -setup %CI_PROJECT_DIR%\%SETUP% || GOTO ErrorExit
+  -setup %CI_PROJECT_DIR%\%SETUP% || GOTO ErrorZipLogfilesAndExit
+
+
+ECHO "Start Artifact Creation"
+TIME /T
+
+CALL :ZipLogfiles
 
 copy "%CYGROOT%\build\coq-local\dev\nsis\*.exe" dev\nsis || GOTO ErrorExit
-7z a coq-opensource-archive-windows-%ARCHLONG%.zip %CYGROOT%\build\tarballs\* || GOTO ErrorExit
+REM The open source archive is only required for release builds
+IF DEFINED WIN_CERTIFICATE_PATH (
+  7z a coq-opensource-archive-windows-%ARCHLONG%.zip %CYGROOT%\build\tarballs\* || GOTO ErrorExit
+) ELSE (
+  REM In non release builds, create a dummy file
+  ECHO "This is not a release build - open source archive not created / uploaded" > coq-opensource-info.txt
+  7z a coq-opensource-archive-windows-%ARCHLONG%.zip coq-opensource-info.txt || GOTO ErrorExit
```

Comment:
Well there is the Readme.txt file in /dev/build/windows, which is quite extensive (I would say) but possibly a bit outdated.

I plan to do a cleanup step (on master), including
- renaming gitlab.bat to ci_windows.bat
- moving the nsis patches into the proper folder
- updating the documentation

Possibly I can move the documentation somewhere else.

Reply:
I looked into places where to document this, but I think it is fine as is. The script uploads a dummy open source archive which contains a text file stating the above echo line. This should be sufficient for everybody to understand why the open source archive is not there. Of cause someone could acidentially upload the dummy, but documentation elsewhere likely wouldn't avoid this either. @maximedenes : In case you have a release checklist, I would put in a line there to check if the open source archive is not a dummy.

Reply:
@maximedenes is not the RM anymore but he will continue to take care of signing the packages. Here is the release checklist: https://github.com/coq/coq/blob/master/dev/doc/release-process.md

Otherwise, documenting this as part of `dev/build/windows/README` sounds good too.

Reply:
I will put this on my todo list. It doesn't really fit into dev/build/windows/README, because this doesn't as yet handle CI at all and this is a CI thing. What is needed is a proper documentation including the windows CI. I plan more changes there (as discussed reuse of libraries from CI) and will document the whole windows CI on the way.